docs/design/queue.md

```diff
@@ -0,0 +1,111 @@
+# Queue
+
+[@k82cn](http://github.com/k82cn); April 17, 2019
+
+## Motivation
+
+`Queue` was introduced in [kube-batch](http://github.com/kubernetes-sigs/kube-batch) long time ago as an internal feature, which makes all jobs are submitted to the same queue, named `default`. As more and more users would like to share resources with each other by queue, this proposal is going to cover primary features of queue achieve that. 
+
+## Function Specificaion
+
+The queue is cluster level, so the user from different namespaces can share resource within a `Queue`. The following section defines the api of queue.
+
+### API
+
+```go
+type Queue struct {
+    metav1.TypeMeta `json:",inline"`
+
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+    // Specification of the desired behavior of a queue
+    // +optional
+    Spec QueueSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+    // Current status of Queue
+    // +optional
+    Status QueueStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+type QueueSpec struct {
+    // The weight of queue to share the resources with each other.
+    Weight int32 `json:"weight,omitempty" protobuf:"bytes,1,opt,name=weight"`
+}
+
+type QueueStatus struct {
+    Unknown int32 `json:"running,omitempty" protobuf:"bytes,1,opt,name=running"`
+    Running int32 `json:"running,omitempty" protobuf:"bytes,2,opt,name=running"`
+    Pending int32 `json:"pending,omitempty" protobuf:"bytes,3,opt,name=pending"`
+    Completed int32 `json:"completed,omitempty" protobuf:"bytes,4,opt,name=completed"`
+    Failed int32 `json:"failed,omitempty" protobuf:"bytes,5,opt,name=failed"`
+    Aborted int32 `json:"aborted,omitempty" protobuf:"bytes,6,opt,name=aborted"`
```

Comment:
What are these statuses used for? For instance, what does a running queue stands for?